File: plusOne.java
public class plusOne {
    /*
    Given a non-empty array of digits representing a non-negative integer, increment one to the integer.

    The digits are stored such that the most significant digit is at the head of the list, and each element in the array contains a single digit.

    You may assume the integer does not contain any leading zero, except the number 0 itself.
     */
    public int[] plusOne(int[] digits) {
        if (digits[digits.length - 1] != 9) {
            digits[digits.length - 1] += 1;
            return digits;
        }
        else {
            int currVal = digits.length - 2;
        }
        return digits;
    }
}

File: PlusOne.py
"""
You are given a large integer represented as an integer array digits, where each digits[i] is the ith digit of the integer. The digits are ordered from most significant to least significant 
in left-to-right order. The large integer does not contain any leading 0's.

Increment the large integer by one and return the resulting array of digits.
"""

class PlusOne(object):
    def plusOne(self, digits):
        index = len(digits) - 1
        carry = False
        digits[index] += 1
        while index >= 0:
            if carry:
                digits[index] += 1
                carry = False
            if digits[index] <= 9:
                return digits
            else:
                digits[index] = digits[index] % 10
                carry = True
                index -= 1
        digits.insert(0, 1)
        return digits